numpy/lib/function_base.py

```diff
@@ -898,15 +898,14 @@ def average(a, axis=None, weights=None, returned=False):
     TypeError: Axis must be specified when shapes of a and weights differ.
 
     """
-    if not isinstance(a, np.matrix):
-        a = np.asarray(a)
+    a = np.asanyarray(a)
 
     if weights is None:
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
         a = a + 0.0
```

Comment:
I like `a  = np.asanyarray(a, dtype=np.result_dtype(a, 'f8'))` because I am worried about the behavior of the `np.multiply(a, wgt)` call. If both `a` and `wgt` happen to be integer arrays we have to worry about overflow etc. 

I like `f8` because that is what `np.mean` explicitly casts to.